torch/_inductor/runtime/benchmarking.py

```diff
@@ -282,6 +352,129 @@ def benchmark_gpu(
         return min(estimated_timing, benchmarked_timing)
 
 
+class GroupedInductorBenchmarker(InductorBenchmarker):
+    def get_interleaved_event_pairs(
+        self: Self, num_callables: int, iters: int
+    ) -> List[List[Tuple[torch.cuda.Event, torch.cuda.Event]]]:
+        """Get `iters` interleaved `num_callables` pairs of CUDA events."""
+        return [self.get_event_pairs(num_callables) for _ in range(iters)]
+
+    def get_interleaved_event_pairs_min_timing(
+        self: Self,
+        interleaved_event_pairs: List[List[Tuple[torch.cuda.Event, torch.cuda.Event]]],
+    ) -> List[float]:
+        """Get the interleaved minimum timings, in milliseconds, for an interleaved
+        grouping of CUDA event pairs.
+        """
+        return [
+            self.get_event_pairs_min_timing(list(event_pairs))
+            for event_pairs in zip(*interleaved_event_pairs)
+        ]
+
+    @time_and_count
+    def benchmark_many_gpu(
+        self: Self,
+        callables: List[Callable[[], Any]],
+        estimation_iters: int = 5,
+        memory_warmup_iters: int = 100,
+        benchmark_iters: int = 100,
+        max_benchmark_duration: int = 25,
+        **kwargs: Any,
+    ) -> List[float]:
+        """Benchmark many GPU callables using a custom benchmarking implementation.
+
+        Arguments:
+        - callables: The callables to benchmark.
+
+        Keyword Arguments:
+        - estimation_iters: The number of iterations to run `_callable` during
+        runtime estimation.
+        - memory_warmup_iters: The number of iterations to flush the L2 cache
+        before benchmarking.
+        - benchmark_iters: The number of iterations to run `_callable` during
+        benchmarking.
+        - max_benchmark_duration: The maximum duration of the benchmarking per
+        callable, in milliseconds. An estimated duration is calculated based on
+        the values of `memory_warmup_iters` and `benchmark_iters`, along with the
+        estimated runtime of `_callable` and various other factors, and we then
+        shrink `benchmark_iters` to fit in the alloted maximum duration.
+        - **kwargs: Additional kwargs that may be passed to the fallback.
+
+        Returns:
+        - The minimum runtime of each callable in `callables`, in milliseconds.
+        """
+        # we don't want any outside errors propagating into benchmarking
+        torch.cuda.synchronize()
```

Comment:
We should use the generic device interface synchronize, as that's what triton do_bench does.